lib/model.js

```diff
@@ -1445,7 +1573,27 @@ class Model {
-   *
+   * @param  {Array<Object>}             [options.cte] An array of hashes of attributes to describe a CTE to be added to the search
+   * @param  {String}                    [options.cte[].name] The name used for the CTE. Every CTE must have a unique name.
+   * @param  {Boolean}                   [options.cte[].useModel=true] If true, the CTE will use the model findAll was called on to base the CTE on
+   * @param  {Model}                     [options.cte[].model] Optional parameter model to base the CTE on
+   * @param  {Array<String>}             [options.cte[].cteAttributes] Extra attributes to give a CTE.
+   * @param  {Boolean}                   [options.cte[].unique=true] Ensure each result of the CTE will be a unique row
+   * @param  {Object}                    [options.cte[].order] sets the order of the recursive results. This order is used incrementally and will not provide a consistent order overall. Cannot be used with   Postgres.
+   * @param  {Number}                    [options.cte[].limit] sets the limit of the recursive results. Cannot be used with Postgres.
+   * @param  {Number}                    [options.cte[].offset] sets the offset of the recursive results. Cannot be used with Postgres.
+   * @param  {Object}                    [options.cte[].initial] A hash of options to describe the initial selection  
+   * @param  {Array<Object|Model>}       [options.cte[].initial.include] An array of associatons to include with a cte intial selection. Follows all the normal include form described above
+   * @param  {Object}                    [options.cte[].initial.where] A hash of attributes to describe your search. Used for the initial selection.
+   * @param  {Object}                    [options.cte[].recursive] A hash of options to describe the recursive selection. This query is run until a where fails 
+   * @param  {String}                    [options.cte[].recursive.next] A string that represents the assocition to recurse through.  
+   * @param  {Array<Object|Model>}       [options.cte[].recursive.include] An array of associatons to include with a cte recursive selection. Follows all the normal include form described above
+   * @param  {Object}                    [options.cte[].recursive.where] A hash of attributes to describe your search. Divided into two subobjects, model and cte, described below.
+   * @param  {Object}                    [options.cte[].recursive.where.model] A hash of options to describe the search with any model attributes on an CTE
+   * @param  {Object}                    [options.cte[].recursive.where.cte] A hash of options to describe the search with any cte attributes on an CTE
+   * @param  {String}                    [options.cteSelect] The name of a CTE to use to base the findAll selection on. The CTE must be based on the findAll model. A join is used on the resutls of the findAll and   the CTE.
```

Comment:
@cbauerme makes sense to me, how i write most of my CTEs but i don't necessarily represent everyone either :D